src/mitol/google_sheets/api.py

```diff
@@ -0,0 +1,318 @@
+"""API for the Sheets app"""
+import datetime
+import json
+import logging
+import os
+import pickle
+from collections import namedtuple
+from urllib.parse import urljoin
+
+import pygsheets
+from django.conf import settings
+from django.core.exceptions import ImproperlyConfigured
+from django.db import transaction
+from google.auth.transport.requests import Request
+from google.oauth2.credentials import Credentials
+from google.oauth2.service_account import Credentials as ServiceAccountCredentials
+from googleapiclient.discovery import build
+
+from mitol.common.utils import now_in_utc
+from mitol.google_sheets.constants import (
+    DEFAULT_GOOGLE_EXPIRE_TIMEDELTA,
+    GOOGLE_API_FILE_WATCH_KIND,
+    GOOGLE_API_NOTIFICATION_TYPE,
+    GOOGLE_TOKEN_URI,
+    REQUIRED_GOOGLE_API_SCOPES,
+)
+from mitol.google_sheets.models import (
+    FileWatchRenewalAttempt,
+    GoogleApiAuth,
+    GoogleFileWatch,
+)
+from mitol.google_sheets.utils import (
+    format_datetime_for_google_timestamp,
+    google_timestamp_to_datetime,
+)
+
+log = logging.getLogger(__name__)
+
+DEV_TOKEN_PATH = "localdev/google.token"
+FileWatchSpec = namedtuple(
+    "FileWatchSpec",
+    ["sheet_metadata", "sheet_file_id", "channel_id", "handler_url", "force"],
+)
+
+
+def get_google_creds_from_pickled_token_file(token_file_path):
+    """
+    Helper method to get valid credentials from a local token file (and refresh as necessary).
+    For dev use only.
+    """
+    with open(token_file_path, "rb") as f:
+        creds = pickle.loads(f.read())
+    if creds and creds.expired and creds.refresh_token:
+        creds.refresh(Request())
+        with open(token_file_path, "wb") as token:
+            pickle.dump(creds, token)
+    if not creds:
+        raise ImproperlyConfigured("Local token file credentials are empty")
+    if not creds.valid:
+        raise ImproperlyConfigured("Local token file credentials are invalid")
+    return creds
+
+
+def get_credentials():
+    """
+    Gets valid Google API client credentials
+
+    Returns:
+        google.oauth2.credentials.Credentials: Credentials to be used by the Google Drive client/pygsheets/etc.
+
+    Raises:
+        ImproperlyConfigured: Raised if no credentials have been configured
+    """
+    if settings.DRIVE_SERVICE_ACCOUNT_CREDS:
+        is_sharing_to_service_account = any(
+            email
+            for email in settings.SHEETS_ADMIN_EMAILS
```

Comment:
Could you namespace this and other settings values like `DRIVE_SERVICE_ACCOUNT_CREDS` and add them to `settings/google_sheets.py`?